gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/testing/RemoteBigQueryHelper.java

```diff
@@ -0,0 +1,156 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Utility to create a remote BigQuery configuration for testing.
+ */
+public class RemoteBigQueryHelper {
+
+  private static final Logger log = Logger.getLogger(RemoteBigQueryHelper.class.getName());
+  private static final String DATASET_NAME_PREFIX = "gcloud_test_dataset_temp_";
+  private final BigQueryOptions options;
+
+  private RemoteBigQueryHelper(BigQueryOptions options) {
+    this.options = options;
+  }
+
+  /**
+   * Returns a {@link BigQueryOptions} object to be used for testing.
+   */
+  public BigQueryOptions options() {
+    return options;
+  }
+
+  /**
+   * Deletes a dataset, even if non-empty.
+   *
+   * @param bigquery the BigQuery service to be used to issue the delete request
+   * @param dataset the dataset to be deleted
+   * @return {@code true} if deletion succeeded, {@code false} if the dataset was not found.
+   * @throws BigQueryException upon failure
+   */
+  public static boolean forceDelete(BigQuery bigquery, String dataset) {
+    return bigquery.delete(dataset, BigQuery.DatasetDeleteOption.deleteContents());
+  }
+
+  /**
+   * Returns a dataset name generated using a random UUID.
+   */
+  public static String generateDatasetName() {
+    return DATASET_NAME_PREFIX + UUID.randomUUID().toString().replace('-', '_');
+  }
+
+  /**
+   * Creates a {@code RemoteBigQueryHelper} object for the given project id and JSON key input
+   * stream.
+   *
+   * @param projectId id of the project to be used for running the tests
+   * @param keyStream input stream for a JSON key
+   * @return A {@code RemoteBigQueryHelper} object for the provided options.
+   * @throws BigQueryHelperException if {@code keyStream} is not a valid JSON key stream
+   */
+  public static RemoteBigQueryHelper create(String projectId, InputStream keyStream)
+      throws BigQueryHelperException {
+    try {
+      BigQueryOptions bigqueryOptions = BigQueryOptions.builder()
+          .authCredentials(AuthCredentials.createForJson(keyStream))
+          .projectId(projectId)
+          .retryParams(retryParams())
+          .connectTimeout(60000)
+          .readTimeout(60000)
+          .build();
+      return new RemoteBigQueryHelper(bigqueryOptions);
+    } catch (IOException ex) {
+      if (log.isLoggable(Level.WARNING)) {
+        log.log(Level.WARNING, ex.getMessage());
+      }
+      throw BigQueryHelperException.translate(ex);
+    }
+  }
+
+  /**
+   * Creates a {@code RemoteBigQueryHelper} object for the given project id and JSON key path.
+   *
+   * @param projectId id of the project to be used for running the tests
+   * @param keyPath path to the JSON key to be used for running the tests
+   * @return A {@code RemoteBigQueryHelper} object for the provided options.
+   * @throws BigQueryHelperException if the file pointed by {@code keyPath} does not exist
+   */
+  public static RemoteBigQueryHelper create(String projectId, String keyPath)
```

Comment:
Just noticed that we also do the same for `Storage`. I still favor not having them (to avoid complaints that it does not work for some people) but not strongly, so I leave it up to you.